addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudConnection.java

```diff
@@ -29,27 +35,36 @@
  *
  * @author Patrik Gfeller - Initial Contribution
  * @author Patrik Gfeller - SOCKET_TIMEOUT changed from 2500 to 10000
+ * @author Martin van Wingerden - add support for custom CA of https://fmipmobile.icloud.com
  */
 public class ICloudConnection {
     private static final String ICLOUD_URL = "https://www.icloud.com";
-    private static final String ICLOUD_API_URL = "https://fmipmobile.icloud.com/fmipservice/device/";
+    private static final String ICLOUD_API_BASE_URL = "https://fmipmobile.icloud.com";
+    private static final String ICLOUD_API_URL = ICLOUD_API_BASE_URL + "/fmipservice/device/";
     private static final String ICLOUD_API_COMMAND_PING_DEVICE = "/playSound";
     private static final String ICLOUD_API_COMMAND_REQUEST_DATA = "/initClient";
-    private static final int SOCKET_TIMEOUT = 10000;
+    private static final int SOCKET_TIMEOUT = 10;
 
     private final Gson gson = new GsonBuilder().create();
     private final String iCloudDataRequest = gson.toJson(ICloudAccountDataRequest.defaultInstance());
 
-    private final Properties httpHeader;
-    private final URL iCloudDataRequestURL;
-    private final URL iCloudFindMyDeviceURL;
+    private final HttpClient httpClient;
+    private final String authorization;
+    private final URI iCloudDataRequestURL;
+    private final URI iCloudFindMyDeviceURL;
 
-    public ICloudConnection(String appleId, String password)
-            throws MalformedURLException, UnsupportedEncodingException {
-        byte[] authorization = Base64.getEncoder().encode((appleId + ":" + password).getBytes());
-        iCloudDataRequestURL = new URL(ICLOUD_API_URL + appleId + ICLOUD_API_COMMAND_REQUEST_DATA);
-        iCloudFindMyDeviceURL = new URL(ICLOUD_API_URL + appleId + ICLOUD_API_COMMAND_PING_DEVICE);
-        httpHeader = createHttpHeader(authorization);
+    public ICloudConnection(HttpClientFactory httpClientFactory, String appleId, String password)
+            throws UnsupportedEncodingException, URISyntaxException {
+        this.httpClient = httpClientFactory.createHttpClient("icloud", ICLOUD_API_BASE_URL);
+        try {
+            this.httpClient.start();
+            // TODO when should I stop it?
```

Comment:
@kaikreuzer would you know this

Reply:
The JavaDoc says: "The client lifecycle should be the same as for your service."
So as the `ICloudConnection` instance is created in `ICloudAccountBridgeHandler.initialize()`, it should be closed (and with it the HTTP client) in `ICloudAccountBridgeHandler.dispose()`.

Reply:
I changed it